### ISSUE TITLE
Participant referral tracking

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#340]
**TL;DR**: A way for families to sign up through a referral link associated with another account, so that we can tell that family A referred family B to Lookit.

**Narrative**
"As a Lookit admin, I would like to encourage word-of-mouth advertising by allowing families to refer other families to Lookit and receive some reward (e.g. a gift card) when the family completes their first study."

**Acceptance Criteria**

- [ ] Each participant account is associated with a unique referral code. (Researcher accounts are not associated with referral links.)
- [ ] In the "your account" section, a link with the referral code is displayed. This link can be shared with others; when someone clicks on this link to get to Lookit and registers for an account, the referral code and/or reference to the referring account is stored.
- [ ] In the "your account" section, the referring family can see how many other families they have referred who have successfully completed at least one study (defined as consent being approved). 
- [ ] The count above is added to the "manage recruitment" view, i.e. we can see a distribution of how many people each family refers.
- [ ] There is a log of successful referrals (referrer, referred, timestamp, referrerRewardSent, referredRewardSent) accessible by a Lookit admin. Ideally this would include (for now) a space to enter a gift card code and a button to send an email with the code to the recipient, for each user.
- [ ] There is an event hook on consent approval that - if referral credit has not already been assigned and a referrer exists, notifies Lookit admins of referrer/referred IDs, creates any necessary objects, and flags that the referral credit has been assigned. 
- [ ] If consent is approved, rejected, and approved again for the first study a referred family does, that counts as one referral, not two. If consent is approved, then rejected, that counts as one referral. I.e., the first time consent is approved is permanent. 
- [ ] If you already have an account and use the referral link to get to Lookit, nothing about your account changes.

**Implementation Notes**
We could use [pinax-referrals](https://github.com/pinax/pinax-referrals) although it may be overkill.

* Probably model referral codes with a separate ReferralToken model that maps user IDs -> unique codes
* Add to urls so we can access the referral code during registration
* Add referrerID, referralCreditAssigned to user model
* May also want to model successfulReferral object (with referrerID, refereeID, timestamp) for display to admin and use in counting up referrals
* We could try to set up something more flexible to track different sorts of actions (rather than a single "you have been successfully referred upon first consent approval") but my impression is it will be substantially more work up front without any clear imagined use case.

**Comment by kimberscott** [comment]
Merged with https://github.com/lookit/lookit-api/issues/622
